src/Aspire.Hosting.AWS/CloudFormation/CloudFormationExtensions.cs

```diff
@@ -0,0 +1,193 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Amazon.CloudFormation;
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.AWS;
+using Aspire.Hosting.AWS.CloudFormation;
+using Aspire.Hosting.Lifecycle;
+using Aspire.Hosting.Publishing;
+using Aspire.Hosting.Utils;
+using Microsoft.Extensions.Logging;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Extension methods for adding AWS CloudFormation as a provisioning resource.
+/// </summary>
+public static class CloudFormationExtensions
+{
+    /// <summary>
+    /// Add a CloudFormation stack for provisioning application resources.
+    /// </summary>
+    /// <param name="builder"></param>
+    /// <param name="stackName">The name of the CloudFormation stack.</param>
+    /// <param name="templatePath">The path to the CloudFormation template that defines the CloudFormation stack.</param>
+    /// <returns></returns>
+    public static IResourceBuilder<ICloudFormationTemplateResource> AddAWSCloudFormationTemplate(this IDistributedApplicationBuilder builder, string stackName, string templatePath)
+    {
+        var resource = new CloudFormationTemplateResource(stackName, templatePath);
+        var cfBuilder = builder.AddResource(resource)
+                                .WithAnnotation(new ManifestPublishingCallbackAnnotation(context => WriteCloudFormationTemplateResourceToManifest(context, resource)));
+
+        builder.Services.TryAddLifecycleHook<CloudFormationLifecycleHook>();
+        return cfBuilder;
+    }
+
+    /// <summary>
+    /// Add parameters to be provided to CloudFormation when creating the stack for the template.
+    /// </summary>
+    /// <param name="builder"></param>
+    /// <param name="parameterName">Name of the CloudFormation parameter.</param>
+    /// <param name="parameterValue">Value of the CloudFormation parameter.</param>
+    /// <returns></returns>
+    public static IResourceBuilder<ICloudFormationResource> AddParameter(this IResourceBuilder<ICloudFormationTemplateResource> builder, string parameterName, string parameterValue)
+    {
+        builder.Resource.AddParameter(parameterName, parameterValue);
+        return builder;
+    }
+
+    /// <summary>
+    /// Add a CloudFormation stack for provisioning application resources.
+    /// </summary>
+    /// <param name="builder"></param>
+    /// <param name="stackName">The name of the CloudFormation stack.</param>
+    /// <returns></returns>
+    public static IResourceBuilder<ICloudFormationStackResource> AddAWSCloudFormationStack(this IDistributedApplicationBuilder builder, string stackName)
+    {
+        var resource = new CloudFormationStackResource(stackName);
+        var cfBuilder = builder.AddResource(resource)
+                                .WithAnnotation(new ManifestPublishingCallbackAnnotation(context => WriteCloudFormationStackResourceToManifest(context, resource)));
```

Comment:
This can be a little less verbose:

```suggestion
                                .WithManifestPublishingCallback(resource.WriteToManifest);
```

Then move WriteToManifest, to `CloudFormationStackResource`. It's a bit cleaner this way.

Reply:
Done